### ISSUE TITLE
material-ui broken by control flow for commonjs exports

### ISSUE THREAD

**Issue by sandersn** [microsoft/TypeScript#43713]
Source:

```js
// @filename: source.js
function isAnnotatedComponent(declaration) {
    return false;
}
const findAnnotatedComponentsResolver = (ast) => {
    return ast;
};

module.exports = findAnnotatedComponentsResolver;
module.exports.isAnnotatedComponent = isAnnotatedComponent;

// @filename: sink.js
const { isAnnotatedComponent } = require('./source');
isAnnotatedComponent
```

Actual: 

Error: Module '"./source"' has no exported member 'isAnnotatedComponent'.

Expected: No Error.

Because of this PR: #42751

This doesn't happen when `findAnnotatedComponentsResolver` is a function declaration, so I suspect some part of the change forgets to look at variable declarations with arrow initialisers.

Found as a break in material-ui.

**Comment by sandersn** [comment]
Note: So far I can't repro in the editor, only from `tsc --noemit --checkjs --skiplibcheck source.js sink.js`
